### PR TITLE
Removed parameter allowPrivateAddresses and host check when node starts

### LINKED ISSUE #3707: Remove host check when node starts
## Overview

Currently, node have a check for valid host when starts. Also there is parameter to `--allow-private-addresses` to allow binding to a local host. 
Both of these doesn't make sense. Host can legitimate be not accessible inside the server where node is running and having additional option for local hosts just makes configuration more difficult.

More information why this is also a problem can be found on this issue.
- #3518

Why is meaningless to check valid host when node starts is because anyone can modify node behavior and set any host in node address as it likes. **Check of valid host should be on the receiver side when node receives request from peers to connect, to prevent local loopback to invalid hosts.**

## Solution

For now we need to remove host checking completely and remove `--allow-private-addresses` config parameter.
https://github.com/rchain/rchain/blob/370eee8f8227f6b8c93892d6a300bcc59f5b7e65/node/src/main/scala/coop/rchain/node/Main.scala#L89

NOTE: keep `--allow-private-addresses` CLI option but print warning to the log that this option is not used anymore and will be removed in future versions, to minimize breaking changes for users which are now forced to use it, like VSCode extension.

- [x] Remove host check when node starts
  #3734
- [x] Implement host check for peers to prevent mixing of public/private subnets
  #3761
  This can pose security risk if public nodes accept local subnets. In that way public nodes can try to probe local networks of connected peers.

### References

- #2474
- https://rchain.atlassian.net/browse/RCHAIN-3435

### PR DESCRIPTION
**Pull request by stanislavlyalin**
## Overview

Also removed related code. A deprecation warning is printed to the output if the user has specified a parameter. This is for compatibility with current clients, such as the VSCode extension, and should be removed in a future releases.

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by nzpr** [comment]
Should **Check of valid host should be on the receiver side when node receives request from peers to connect, to prevent local loopback to invalid hosts.** from the issue be done in this PR?

File: node/src/test/scala/coop/rchain/node/configuration/commandline/ConfigMapperSpec.scala [hunk 1]
```diff
@@ -131,7 +131,6 @@ class ConfigMapperSpec extends AnyFunSuite with Matchers {
       protocolServer = ProtocolServer(
         networkId = "testnet",
         host = Some("localhost"),
-        allowPrivateAddresses = true,
```
Comment (tgrospic) [inline node/src/test/scala/coop/rchain/node/configuration/commandline/ConfigMapperSpec.scala:134]: Please also search for other places where string `allow-private-addresses` appears to remove it like up in this file.
https://github.com/rchain/rchain/blob/3110814fdacc27469abc38e0c19756278ad77b21/node/src/test/scala/coop/rchain/node/configuration/commandline/ConfigMapperSpec.scala#L32

Comment (stanislavlyalin) [reply node/src/test/scala/coop/rchain/node/configuration/commandline/ConfigMapperSpec.scala:134]: [Done](https://github.com/rchain/rchain/pull/3734/commits/4588bd268e8f283703061998e38db907ceff5e9e).

**Review by nzpr — APPROVED** [review]
I suggest we go ahead and merge this, https://github.com/rchain/rchain/pull/3734#issuecomment-1154430048 this should be addressed in future PR, @maximus19910112 can investigate the issue more

**Review by tgrospic — APPROVED** [review]
LGTM

**Comment by stanislavlyalin** [comment]
bors merge

**Pull request merged**